internal/vcs/git/commits.go

```
@@ -285,6 +285,19 @@ func isBadObjectErr(output, obj string) bool {
 //
 // The caller is responsible for doing checkSpecArgSafety on opt.Head and opt.Base.
 func commitLog(ctx context.Context, repo api.RepoName, opt CommitsOptions, checker authz.SubRepoPermissionChecker) (commits []*gitdomain.Commit, err error) {
+	wrappedCommits, err := getWrappedCommits(ctx, repo, opt)
+	if err != nil {
+		return nil, err
+	}
+	filtered, err := filterCommits(ctx, wrappedCommits, repo, checker)
```

We should fail early here if `err` is not nil.